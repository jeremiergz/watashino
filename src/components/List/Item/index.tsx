import FlexBox, { FlexBoxProps } from 'components/primitives/FlexBox';
import React from 'react';

const Item: React.FC<ItemProps> = ({ children, ...rest }) => {
  return (
    <FlexBox
      alignItems="center"
      as="li"
      fontSize={24}
      fontWeight="semiBold"
      justifyContent="left"
      paddingX={0}
      paddingY={2}
      textAlign="left"
      {...rest}
    >
      {children}
    </FlexBox>
  );
};

Item.displayName = 'ListItem';

export type ItemProps = FlexBoxProps;
export default Item;
