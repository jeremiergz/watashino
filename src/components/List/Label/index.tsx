import Text, { TextProps } from 'components/primitives/Text';
import React from 'react';

const Label: React.FC<LabelProps> = ({ children, ...rest }) => {
  return (
    <Text
      as="label"
      color="secondary"
      display="block"
      fontSize={32}
      fontWeight="semiBold"
      marginBottom={4}
      marginTop={5}
      marginX={0}
      {...rest}
    >
      {children}
    </Text>
  );
};

Label.displayName = 'ListLabel';

export type LabelProps = TextProps;
export default Label;
