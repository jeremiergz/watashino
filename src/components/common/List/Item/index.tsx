import React from 'react';
import Flex, { FlexProps } from '../../../primitives/Flex';

type ItemProps = FlexProps;

const Item = ({ children, ...rest }: ItemProps) => {
  return (
    <Flex
      alignItems="center"
      as="li"
      fontSize={24}
      fontWeight="semi-bold"
      justifyContent="left"
      paddingX={0}
      paddingY={2}
      textAlign="left"
      {...rest}
    >
      {children}
    </Flex>
  );
};

Item.displayName = 'ListItem';

export { ItemProps };
export default Item;
