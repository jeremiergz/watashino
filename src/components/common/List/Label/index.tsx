import React, { FunctionComponent } from 'react';
import Text, { TextProps } from '../../../primitives/Text';

const Label: FunctionComponent<LabelProps> = ({ children, ...rest }) => {
  return (
    <Text
      as="label"
      color="secondary"
      display="block"
      fontSize={32}
      fontWeight="semi-bold"
      marginBottom={4}
      marginTop={5}
      marginX={0}
      {...rest}
    >
      {children}
    </Text>
  );
};

Label.displayName = 'ListLabel';

export type LabelProps = TextProps;
export default Label;
