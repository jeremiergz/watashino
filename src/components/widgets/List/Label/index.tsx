import React from 'react';
import Text, { TextProps } from '../../../primitives/Text';

type LabelProps = TextProps;

const Label = ({ children, ...rest }: LabelProps) => {
    return (
        <Text
            as="label"
            color="secondary"
            display="block"
            fontSize={32}
            fontWeight="semi-bold"
            marginBottom={4}
            marginTop={5}
            marginX={0}
            {...rest}
        >
            {children}
        </Text>
    );
};

Label.displayName = 'ListLabel';

export { LabelProps };
export default Label;
