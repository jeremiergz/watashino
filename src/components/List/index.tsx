import Box, { BoxProps } from 'components/primitives/Box';
import React from 'react';
import Item from './Item';
import Label from './Label';

const List: React.FC<ListProps> & { Item: typeof Item; Label: typeof Label } = ({ children, ...rest }) => {
  return (
    <Box as="ul" listStyle="none" margin={0} padding={0} {...rest}>
      {children}
    </Box>
  );
};

List.displayName = 'List';
List.Item = Item;
List.Label = Label;

export type ListProps = BoxProps;
export default List;
