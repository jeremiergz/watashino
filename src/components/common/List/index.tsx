import React, { FunctionComponent } from 'react';
import Box, { BoxProps } from '../../primitives/Box';
import Item from './Item';
import Label from './Label';

const List: FunctionComponent<ListProps> & { Item: typeof Item; Label: typeof Label } = ({ children, ...rest }) => {
  return (
    <Box as="ul" listStyle="none" margin={0} padding={0} {...rest}>
      {children}
    </Box>
  );
};

List.displayName = 'List';
List.Item = Item;
List.Label = Label;

export type ListProps = BoxProps;
export default List;
