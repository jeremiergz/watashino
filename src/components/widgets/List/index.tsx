import React from 'react';
import styled from 'styled-components';
import BaseBox, { BoxProps } from '../../primitives/Box';
import Item from './Item';
import Label from './Label';

type ListProps = BoxProps;

const Box = styled(BaseBox)`
    list-style: none;
`;

const List = ({ children, ...rest }: ListProps) => {
    return (
        <Box as="ul" margin={0} padding={0} {...rest}>
            {children}
        </Box>
    );
};

List.Label = Label;
List.Item = Item;

export { ListProps };
export default List;
