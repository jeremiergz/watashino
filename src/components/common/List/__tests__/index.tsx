import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import List from '../../List';

describe('components/widgets/List Test Suite', () => {
    it('matches snapshot', () => {
        const jsx = (
            <List>
                <List.Label>List Label</List.Label>
                <List.Item>List Item 0</List.Item>
                <List.Item>List Item 1</List.Item>
            </List>
        );
        expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
    });
});
