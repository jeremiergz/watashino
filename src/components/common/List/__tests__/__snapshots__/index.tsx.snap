// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/List Test Suite matches snapshot 1`] = `
<ul
  className="sc-AykKC sc-AykKE jhrrfD"
>
  <label
    className="sc-AykKD bSeRXq"
    color="secondary"
    display="block"
    fontSize={32}
    fontWeight="semi-bold"
  >
    List Label
  </label>
  <li
    className="sc-AykKC ikGVWR"
    display="flex"
    fontSize={24}
    fontWeight="semi-bold"
  >
    List Item 0
  </li>
  <li
    className="sc-AykKC ikGVWR"
    display="flex"
    fontSize={24}
    fontWeight="semi-bold"
  >
    List Item 1
  </li>
</ul>
`;
